File: backend/BowlerAPI/Data/Bowler.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BowlerAPI.Data;

public partial class Bowler
{
    [Key]
    [Required]
    public int BowlerId { get; set; }

    public string? BowlerLastName { get; set; }

    public string? BowlerFirstName { get; set; }

    public string? BowlerMiddleInit { get; set; }

    public string? BowlerAddress { get; set; }

    public string? BowlerCity { get; set; }

    public string? BowlerState { get; set; }

    public string? BowlerZip { get; set; }

    public string? BowlerPhoneNumber { get; set; }
    [ForeignKey("TeamId")]
    public int? TeamId { get; set; }

    public virtual ICollection<BowlerScore> BowlerScores { get; set; } = new List<BowlerScore>();

    public virtual Team? Team { get; set; }

}

File: frontend1/src/types/Bowler.ts
import { BowlerScores } from './BowlerScore';
import { Team } from './Team';

export type Bowler = {
  bowlerId: number;

  bowlerLastName: string;

  bowlerFirstName: string;

  bowlerMiddleInit: string;

  bowlerAddress: string;

  bowlerCity: string;

  bowlerState: string;

  bowlerZip: string;

  bowlerPhoneNumber: string;
  teamId?: number;
  bowlerScores: BowlerScores[];
  team?: Team;
};

// interface BowlerScores {
//   matchId: number;
//   gameNumber: number;
//   bowlerId: number;
//   rawScore: number;
//   handiCapScore: number;
//   wonGame: boolean;
//   bowler: Bowler;
// }

// interface Team {
//   teanId: number;
//   teamName: string;
//   captainId: number;
//   bowlers: Bowler[];
//   tourneyMatchEvenLaneTeams: TourneyMatch[];
//   tourneyMatchOddLaneTeams: TourneyMatch[];
// }

// interface TourneyMatch {
//   matchId: number;
//   tourneyId: number;
//   lanes: string;
//   oddLaneTeamId: number;
//   evenLaneTeamId: number;
//   evenLaneTeam: Team;
//   matchGames: MatchGame[];
//   oddLaneTeam: Team;
//   tourney: Tournament;
// }

// interface MatchGame {
//   matchId: number;
//   gameNumber: number;
//   winningTeamId: number;
//   match: TourneyMatch;
// }

// interface Tournament {
//   tourneyId: number;
//   tourneyDate: Date;
//   tourneyLocation: string;
//   tourneyMatches: TourneyMatch;
// }
